src/sqlparser/src/parser_v2/expr.rs

```diff
@@ -58,3 +59,91 @@ where
 
     trace("expr_case", parse).parse_next(input)
 }
+
+/// Consome a SQL CAST function e.g. `CAST(expr AS FLOAT)`
+pub fn expr_cast<S>(input: &mut S) -> PResult<Expr>
+where
+    S: TokenStream,
+{
+    let parse = (
+        cut_err(Token::LParen),
+        cut_err(expr).map(Box::new),
+        cut_err(Keyword::AS),
+        cut_err(data_type),
+        cut_err(Token::RParen),
+    )
+        .map(|(_, expr, _, data_type, _)| Expr::Cast { expr, data_type });
```

Comment:
This can also be written using `seq!`. which one do you think is better?

```suggestion
    use Expr:Cast;
    seq! {
        Cast {
            _: Token::LParen,
            expr: expr.map(Box::new),
            _: Keyword::AS,
            data_type: data_type,
            _: Token::RParen,
        }
    }
```

Reply:
I want to know why `seq!` does not support type with multiple segments.

If I can use `Expr::Cast` here, `seq!` is better.

Reply:
https://github.com/winnow-rs/winnow/issues/531

When they resolve this, I'll submit a PR to use `seq!` in these places.

Reply:
https://github.com/winnow-rs/winnow/issues/532